In [1]:
import pandas as pd
import numpy as np

from sklearn.datasets import load_boston
boston_house_prices = load_boston()

boston_house_prices.keys()

dict_keys(['data', 'target', 'feature_names', 'DESCR', 'filename'])

In [2]:
data = boston_house_prices["data"]
X = pd.DataFrame(data, columns=boston_house_prices.feature_names)

target = boston_house_prices["target"]
y = pd.DataFrame(target, columns=["price"])

In [3]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [4]:
from sklearn.ensemble import RandomForestRegressor

model = RandomForestRegressor(n_estimators=1000, max_depth=12, random_state=42)

model.fit(X_train, y_train.values[:, 0])

RandomForestRegressor(max_depth=12, n_estimators=1000, random_state=42)

In [5]:
y_pred_model = model.predict(X_test)
from sklearn.metrics import r2_score
r2_2 = r2_score(y_test, y_pred_model)

r2_2

0.87472606157312

r2 = 0.7112260057484948 (c помощью LinearRegression)
r2_2 = 0.87472606157312 (с помощью RandomForestRegressor)

Модель созданная с помощью RandomForestRegressor показала более точный результат.